ReText/tab.py

```diff
@@ -444,15 +444,76 @@ def openSourceFile(self, fileToOpen):
 		"""Finds and opens the source file for link target fileToOpen.
 
 		When links like [test](test) are clicked, the file test.md is opened.
+		# TODO: broken objective? See TODO below.
 		It has to be located next to the current opened file.
 		Relative paths like [test](../test) or [test](folder/test) are also possible.
+
+		:param fileToOpen:	Link target
+							This can be a file of any extension or a path
+		:returns: 			Filename successfully handled to openFileWrapper
+							or 'noSuchFile' to stop further processing.
+		:rtype: 			str
 		"""
 		if self.fileName:
+			# Please give description of this condition.
+			# (e.g. "if internal link was clicked from within valid open document in a tab"?)
 			currentExt = splitext(self.fileName)[1]
 			basename, ext = splitext(fileToOpen)
 			if ext in ('.html', '') and exists(basename + currentExt):
+				# TODO: It seems this should catch markdown links like
+				#  [test](test) but than ext not in ('.html', '')
 				self.p.openFileWrapper(basename + currentExt)
 				return basename + currentExt
-		if exists(fileToOpen) and get_markup_for_file_name(fileToOpen, return_class=True):
+
+		if get_markup_for_file_name(fileToOpen, return_class=True):
+			if not QFile.exists(fileToOpen):
+				# ReText should only support to create markup-files, no plain
+				# text files. Thereby it facilitates writing cross-linked
+				# documentation.
+				# Consequently create-file functionality can be nested here.
+				if not self.createFileQuestion(fileToOpen):
+					# File still not exists
+					return 'noSuchFile'
```

Comment:
What if the file name is `noSuchFile`? Maybe better to `raise FileNotFoundError`?

Reply:
Tested raising an error but that started real trouble. My intention was to return 'something' to trigger the return-statement in (webkit/webengine)preview.py

My understanding is that the returnvalue is never processed and [return fileToOpen] could be replaced with [return 'filesuccessfullyopened'] without breaking something.